crates/egui/src/context.rs

```diff
@@ -411,16 +411,18 @@ impl Context {
             let mut slf = self.write();
 
             slf.layer_rects_this_frame
-                .entry(layer_id)
+                .entry(layer.area_layer)
                 .or_default()
-                .push((id, interact_rect));
+                .push((id, layer.z, interact_rect));
 
             if hovered {
                 let pointer_pos = slf.input.pointer.interact_pos();
                 if let Some(pointer_pos) = pointer_pos {
-                    if let Some(rects) = slf.layer_rects_prev_frame.get(&layer_id) {
-                        for &(prev_id, prev_rect) in rects.iter().rev() {
-                            if prev_id == id {
+                    if let Some(rects) = slf.layer_rects_prev_frame.get_mut(&layer.area_layer) {
+                        rects.sort_by_key(|(_id, z, ..)| *z);
```

Comment:
re-sorting this on each interaction is unnecessary. Better to sort once at the end of each frame instead.